syntax = "proto3";

package grpcarch;

option go_package = "github.com/grpcarchitecture/proto";
option csharp_namespace = "GrpcArchitecture.Proto";

// Common request metadata for tracing context
message RequestMetadata {
  string request_id = 1;
  string trace_id = 2;
  string caller_service = 3;
  int64 timestamp_ms = 4;
}

// Common response status
message ResponseStatus {
  bool success = 1;
  string message = 2;
  int32 error_code = 3;
}

// Generic data payload
message DataPayload {
  string id = 1;
  string content = 2;
  map<string, string> attributes = 3;
}
